src/chatClient.cpp

```diff
@@ -3546,43 +3546,38 @@ void GroupChatRoom::enablePreview(uint64_t ph)
 
 bool GroupChatRoom::publicChat() const
 {
+    assert(mChat);
     if (mChat)
     {
         return (mChat->crypto()->isPublicChat());
     }
 
+    parent.mKarereClient.api.callIgnoreResult(&::mega::MegaApi::sendEvent, 99011, "GroupChatRoom::publicChat(), chatd::Chat isn't yet created");
+
     return false;
 }
 
 uint64_t GroupChatRoom::getPublicHandle() const
 {
+    assert(mChat);
     if (mChat)
     {
         return (mChat->getPublicHandle());
     }
 
+    parent.mKarereClient.api.callIgnoreResult(&::mega::MegaApi::sendEvent, 99011, "GroupChatRoom::getPublicHandle(), chatd::Chat isn't yet created");
```

Comment:
For the same event id, the stats server only records the name of the first occurrence. In result, you will not be able to differentiate whether the event comes from `getPublicHandle()` or from `publicChat()`. I think that using only one event is enough, anyway :)

Reply:
Since the caller takes ownership, they need the same address than `new []` returned. Otherwise `delete []` will result on undefined behavior.
In short, disregard my suggestion. Thanks